gempython/tools/amc_user_functions_xhal.py

```diff
@@ -185,6 +185,12 @@ def __init__(self, cardName, debug=False, gemType="ge11"):
         else:
             raise KeyError("Unrecognized gemType {0}".format(gemType))
 
+        #Determine the number of VFATs per geb based on the gemType
+        if gemType in vfatsPerGemVariant.keys():
+            self.NVFAT = vfatsPerGemVariant[gemType]
```

Comment:
we won't have a situation where `detType` has a different number of VFATs correct?
e.g. {M1,...,M8} all have 12 VFATs?

Reply:
Also in `HwOptohybrid` this is `self.nVFATs` vs. `self.NVFAT`.

Should we keep both these data members...in retrospect probably the better way to go would have been to have:

`HwOptohybrid` inherit from `HwAMC` and `HwVFAT` inherit from `HwOptohybrid`.  Then all this `parentX` crap which is, admittedly, messy could disappear.  Although whether that type of change is appropriate for this PR (or ever given this is #legacy) is unclear.  Alas...my programming skills have improved since the original design...

Reply:
I would caution against making API changes at this point, since this legacy stuff is supposed to be frozen anyway...
We will have to potentially think of how to map this in the updated API, where currently there is not going to *be* a `HwVFAT`, though it could be added if deemed absolutely critical (would just have to think of how best to do so...)

Reply:
> I would caution against making API changes at this point, since this legacy stuff is supposed to be frozen anyway...

Indeed.  For this PR then I would prefer we just standardized the data member name; so it should be either `self.NVFAT` or `self.nVFATs` with a slight preference for `self.nVFATs` simply because it pre-dates this PR in `HwOptohybrid`.